weakmap.c

```diff
@@ -12,12 +12,47 @@ struct weakmap {
     VALUE final;
 };
 
+static int
+wmap_replace_ref(st_data_t *key, st_data_t *value, st_data_t _argp, int existing)
+{
+    *key = rb_gc_location((VALUE)*key);
+
+    VALUE *values = (VALUE *)value;
+    VALUE size = (long long)values[0];
+
+    for (VALUE index = 1; index < size; index++) {
```

Comment:
Shouldn't this be `index <= size`, since we're starting `index` at 1?

Reply:
Yup, you are right